provider/provider.go

```diff
@@ -50,3 +50,25 @@ func EnsureTrailingDot(hostname string) string {
 
 	return strings.TrimSuffix(hostname, ".") + "."
 }
+
+// Tells which entries need to be respectively
+// added, removed, or left untouched for "current" to be transformed to "desired"
+func Difference(current, desired []string) (add []string, remove []string, leave []string) {
```

Comment:
This is mostly personal preference, but I'd rather avoid using named return values. I think we tried to avoid them across most of the codebase and I would appreciate sticking to that.